web/packages/shared/utils/highbar.ts

```diff
@@ -60,33 +60,55 @@ export function mergeDeep(target: MergeTarget, ...sources: Array<MergeTarget>) {
   return mergeDeep(target, ...sources);
 }
 
-type CompareArray = Array<Record<string, unknown>>;
+/** Recursively freezes an object and its contents. */
+export function freezeDeep<T extends object>(obj: T): T {
+  // Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze
+  // Any copyright is dedicated to the Public Domain:
+  // https://creativecommons.org/publicdomain/zero/1.0/
 
-export function arrayObjectIsEqual(
-  arr1: CompareArray,
-  arr2: CompareArray
-): boolean {
-  const compareArrays = (arr1, arr2) =>
-    arr1.length === arr2.length &&
-    arr1.every((obj, idx) => compareObjects(obj, arr2[idx]));
+  // Retrieve the property names defined on object
+  const propNames = Reflect.ownKeys(obj);
```

Comment:
nit: spell out because `prop` may be confusing if this is referring to react props. 
```suggestion
  const objectPropertyNames = Reflect.ownKeys(obj);
```